src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -133,7 +134,7 @@ public EC2FleetCloud(final String awsCredentialsId,
                          final Integer minSize,
                          final Integer maxSize,
                          final Integer numExecutors) {
-        super(FLEET_CLOUD_ID);
+        super(name == null || name.equals("") ? FLEET_CLOUD_ID : name);
```

Comment:
According to ```Cloud.java``` javadoc name should be:

```
/**
     * Uniquely identifies this {@link Cloud} instance among other instances in {@link jenkins.model.Jenkins#clouds}.
     *
     * This is expected to be short ID-like string that does not contain any character unsafe as variable name or
     * URL path token.
     */
```

Reply:
I think we need to add validation or if we just want to show nice name in UI we can override ```Cloud#getDisplayName```

Reply:
Hi @terma, 
I have looked at some of the popular cloud plugins and did not find any validation for cloud name within them:
* [kubernetes-plugin](https://github.com/jenkinsci/kubernetes-plugin/blob/master/src/main/java/org/csanchez/jenkins/plugins/kubernetes/KubernetesCloud.java#L142)
* [docker-plugin](https://github.com/jenkinsci/docker-plugin/blob/master/src/main/java/com/nirima/jenkins/plugins/docker/DockerCloud.java#L127)
* [ec2-plugin](https://github.com/jenkinsci/ec2-plugin/blob/master/src/main/java/hudson/plugins/ec2/EC2Cloud.java#L171)

I believe that if a validation is needed, it should happen in the super class and not done by every  plugin separately.

As for the comment about overriding the `Cloud#getDisplayName`, the reasoning for specifying a name, is for being able to have multiple clouds and distinguish between them, not just from the UI, but from configuration automation such as configuration as code or groovy init scripts.

Hope I made myself clear...
Cheers!